### PR TITLE
Bluetooth: testlib: Create `testlib/conn.h`

### PR DESCRIPTION
**Pull request by alwa-nordic**
This header file declares the functions that rely on `bluetooth/conn.h`.

This collects `testlib/connect.h`, `conn_ref.h` and `conn_wait.h` into one.

Initial documentation for the functions in this file is also added.

### CODE REVIEW

File: tests/bsim/bluetooth/host/att/retry_on_sec_err/client/main.c [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #include "../common_defs.h"
 #include "../test_utils.h"
 
-#include "testlib/connect.h"
```
Comment (jori-nordic) [inline tests/bsim/bluetooth/host/att/retry_on_sec_err/client/main.c:16]: why the switch from " to < ?

**Review by jori-nordic — DISMISSED** [review]

File: tests/bsim/bluetooth/host/att/retry_on_sec_err/client/main.c (hunk 1 above)
Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/att/retry_on_sec_err/client/main.c:16]: No reason

**Review by theob-pro — DISMISSED** [review]

**Comment by carlescufi** [comment]
@dcpleung @andyross @jhedberg could you review the changes in the public atomic operations headers?

**Review by dcpleung — COMMENTED** [review]
LGTM. Seems reasonable.

File: include/zephyr/sys/atomic.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
  * Copyright (c) 1997-2015, Wind River Systems, Inc.
  * Copyright (c) 2021 Intel Corporation
- *
+ * Copyright (c) 2023 Nordic Semiconductor ASA
  * SPDX-License-Identifier: Apache-2.0
```
Comment (jhedberg) [inline include/zephyr/sys/atomic.h:5]: Is there a reason you're changing the formatting here by removing the empty line between the last Copyright statement and the SPDX line? I did a quick grep through the Zephyr tree, and it seems having an empty line in between these is by far the most common way to format this.

File: include/zephyr/sys/atomic_arch.h [hunk 3]
```diff
@@ -1,12 +1,15 @@
 /*
  * Copyright (c) 2021 Demant A/S
- *
+ * Copyright (c) 2023 Nordic Semiconductor ASA
  * SPDX-License-Identifier: Apache-2.0
```
Comment (jhedberg) [inline include/zephyr/sys/atomic_arch.h:4]: Same here

File: include/zephyr/sys/atomic_builtin.h [hunk 4]
```diff
@@ -2,13 +2,16 @@
 
 /*
  * Copyright (c) 1997-2015, Wind River Systems, Inc.
- *
+ * Copyright (c) 2023 Nordic Semiconductor ASA
  * SPDX-License-Identifier: Apache-2.0
```
Comment (jhedberg) [inline include/zephyr/sys/atomic_builtin.h:6]: And here

File: include/zephyr/sys/atomic_types.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 1997-2015, Wind River Systems, Inc.
+ * Copyright (c) 2021 Intel Corporation
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (jhedberg) [inline include/zephyr/sys/atomic_types.h:4]: And here

File: include/zephyr/sys/atomic.h (hunk 2 above)
Comment (alwa-nordic) [reply include/zephyr/sys/atomic.h:5]: It takes less space. Are you nak-ing the PR over this?

Comment (alwa-nordic) [reply include/zephyr/sys/atomic.h]: Fixed

File: include/zephyr/sys/atomic_types.h (hunk 5 above)
Comment (alwa-nordic) [reply include/zephyr/sys/atomic_types.h]: Fixed

**Review by jhedberg — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]

**Pull request merged**